internal/operators/odf/validations.go

```diff
@@ -78,20 +77,13 @@ func (o *operator) computeResourcesAllNodes(cluster *models.Cluster, odfClusterR
 		*/
 
 		role := common.GetEffectiveRole(host)
-		if !isCompactMode {
-			if role == models.HostRoleAutoAssign {
-				status = "For ODF Standard Mode, all host roles must be assigned to master or worker."
-				err = errors.New("Role is set to auto-assign for host ")
-				return status, err
-			}
-			if role == models.HostRoleWorker {
-				status, err = o.computeNodeResourceUtil(host, odfClusterResources, standardMode)
-				if err != nil {
-					return status, err
-				}
-			}
-		} else {
-			status, err = o.computeNodeResourceUtil(host, odfClusterResources, compactMode)
+		if role == models.HostRoleAutoAssign && !isCompactMode {
+			status = "For ODF Standard Mode, all host roles must be assigned to master or worker."
+			err = errors.New("Role is set to auto-assign for host ")
+			return status, err
+		}
+		if role == models.HostRoleWorker || isCompactMode {
```

Comment:
I think there's a difference between using the cluster to determine if we're standard or compact vs using the number of hosts.

It could be that this causes a difference in behavior if a cluster is set to HA, but we don't actually have the correct number of hosts yet, right?

Reply:
Oh was this only previously used for logging? Maybe it's not an issue, but worth confirming.

Reply:
To be honest I've just refactored the code to a simpler non-duplicated one and the conditions should be the same. Nevertheless if we spot a bug I can take advantage of this PR to fix it

Reply:
I think the confusing thing is what these conditions mean.

Maybe we can create small functions that call out what we're actually doing then this could look like:
`if hostWillRunODF(host, mode) {`

```
func hostWillRunODF(...) bool {
  return host.role == models.HostRoleWorker || mode == CompactMode
}
```